presto-main/src/main/java/com/facebook/presto/sql/planner/SubqueryPlanner.java

```diff
@@ -291,6 +294,33 @@ private PlanBuilder appendQuantifiedComparisonApplyNode(PlanBuilder subPlan, Qua
             // given subquery is already appended
             return subPlan;
         }
+        switch (quantifiedComparison.getComparisonType()) {
+            case EQUAL:
+                switch (quantifiedComparison.getQuantifier()) {
+                    case ANY:
+                    case SOME:
+                        // A = ANY B <=> A IN B
+                        InPredicate inPredicate = new InPredicate(quantifiedComparison.getValue(), quantifiedComparison.getSubquery());
+                        subPlan.getTranslations().addIntermediateMapping(quantifiedComparison, inPredicate);
+                        return appendInPredicateApplyNode(subPlan, inPredicate, correlationAllowed);
+                }
+                break;
+
+            case NOT_EQUAL:
+                switch (quantifiedComparison.getQuantifier()) {
+                    case ALL:
+                        // A <> ALL B <=> !(A IN B) <=> !(A = ANY B)
+                        QuantifiedComparisonExpression rewrittenAny = new QuantifiedComparisonExpression(
+                                ComparisonExpressionType.EQUAL,
+                                Quantifier.ANY,
+                                quantifiedComparison.getValue(),
+                                quantifiedComparison.getSubquery());
+                        Expression notAny = new NotExpression(rewrittenAny);
+                        subPlan.getTranslations().addIntermediateMapping(quantifiedComparison, notAny);
+                        return appendQuantifiedComparisonApplyNode(subPlan, rewrittenAny, correlationAllowed);
```

Comment:
Is `rewrittenAny` in this line intentional? Or did you mean `notAny`? If it's a mistake, make sure to add a test that reproduces the issue.